### PR TITLE
[#4270] New API plugin - Fetch info about an iRODS file descriptor (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: plugins/api/src/get_file_descriptor_info.cpp [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    }
+
+    auto to_bytes_buffer(const std::string& _s) -> bytesBuf_t*
+    {
+        const auto buf_size = _s.length() + 1;
+
+        char* buf = new char[buf_size]{};
+        std::strncpy(buf, _s.c_str(), _s.length());
+
+        bytesBuf_t* bbp = new bytesBuf_t{};
```
Comment (trel) [inline plugins/api/src/get_file_descriptor_info.cpp]: Codacy Issue found: [Uninitialized variable: bbp](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5176159)

**Comment by trel** [comment]
i looked hard - couldn't find anything to complain about.     codacy found a thing, but pretty sure it's wrong.  # when ready.

File: server/api/include/rs_get_file_descriptor_info.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * \fn int rs_get_file_descriptor_info(rsComm_t* _comm, const char* _json_input, char* _json_output)
+ *
+ * \brief Returns information about an iRODS file descriptor.
+ *
+ * The caller is expected to pass the same rsComm_t* used to open the data object.
+ *
+ * \user Server
+ *
+ * \since 4.3.0
```
Comment (korydraughn) [inline server/api/include/rs_get_file_descriptor_info.hpp]: wrong version

Comment (korydraughn) [reply server/api/include/rs_get_file_descriptor_info.hpp]: fixed.

**Pull request merged**